src/webgpu/util/texture/color_space_conversions.spec.ts

```diff
@@ -0,0 +1,108 @@
+export const description = 'Color space conversion helpers';
+
+import { Fixture } from '../../../common/framework/fixture.js';
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+import { ErrorWithExtra } from '../../../common/util/util.js';
+import { makeInPlaceColorConversion } from '../color_space_conversion.js';
+import { clamp } from '../math.js';
+
+import { TexelView } from './texel_view.js';
+import { findFailedPixels } from './texture_ok.js';
+
+const kTestColors = [
+  [0xff, 0, 0],
+  [0, 0xff, 0],
+  [0, 0, 0xff],
+  [0x80, 0x80, 0],
+  [0, 0x80, 0x80],
+  [0x80, 0, 0x80],
+] as const;
+
+function floatToU8(v: number) {
+  return clamp(Math.round(v * 255), { min: 0, max: 255 });
+}
+
+export const g = makeTestGroup(Fixture);
+
+g.test('color_space_conversion')
+  .desc(`Test color space conversion helpers matches canvas 2d's color space conversion`)
+  .params(u =>
+    u.combineWithParams([
+      { srcColorSpace: 'srgb', dstColorSpace: 'display-p3' },
+      { srcColorSpace: 'display-p3', dstColorSpace: 'srgb' },
+    ] as { srcColorSpace: PredefinedColorSpace; dstColorSpace: PredefinedColorSpace }[])
+  )
+  .fn(t => {
+    const { srcColorSpace, dstColorSpace } = t.params;
+
+    // putImageData an ImageData(srcColorSpace) in to a canvas2D(dstColorSpace)
+    // then call getImageData. This will convert the colors via the canvas 2D API
+    const width = kTestColors.length;
+    const height = 1;
+    const imgData = new ImageData(
+      new Uint8ClampedArray(kTestColors.map(v => [...v, 255]).flat()),
+      width,
+      height,
+      { colorSpace: srcColorSpace }
+    );
+    const ctx = new OffscreenCanvas(width, height).getContext('2d', {
+      colorSpace: dstColorSpace,
+    })!;
+    ctx.putImageData(imgData, 0, 0);
+    const expectedData = ctx.getImageData(0, 0, width, height).data;
+
+    const conversionFn = makeInPlaceColorConversion({
+      srcPremultiplied: false,
+      dstPremultiplied: false,
+      srcColorSpace,
+      dstColorSpace,
+    });
+
+    // Convert the data via our conversion functions
+    const convertedData = new Uint8ClampedArray(
```

Comment:
(Not a comment Just for curious). 
Would you mind to explain a bit about how this test could detect the behaviour change from `=` assignment to `Object.assign` ?

Reply:
It detects it because `makeInPlaceColorConversion`, at least by name, is supposed to make an object that is updated "in place"

```
          const floatColor = { R, G, B, A: 1 };          
          conversionFn(floatColor);
          return [
            floatToU8(floatColor.R),
            floatToU8(floatColor.G),
            floatToU8(floatColor.B),
            floatToU8(floatColor.A),
          ];
```

Expects that `floatColor` is updated. In the assignment (`=`) version, a new object is created inside the conversion function, That object is then discarded. The function out here is still looking at `floatColor`.

In order to work as it was, this code would need to change to this

```
          const floatColor = { R, G, B, A: 1 };          
          const convertedColor = conversionFn(floatColor);
          return [
            floatToU8(convertedColor.R),
            floatToU8(convertedColor.G),
            floatToU8(convertedColor.B),
            floatToU8(convertedColor.A),
          ];
```

I'm happy to change it to this style. I was just going from the momentum given that the function is called `makeInPlaceColorConversion`